src/main/notifications/index.ts

```diff
@@ -27,6 +27,11 @@ class NotificationManager {
     private upgradeNotification?: NewVersionNotification;
     private restartToUpgradeNotification?: UpgradeNotification;
 
+    constructor() {
+        // Run this very early to perform an early permission check
+        getDoNotDisturb();
```

Comment:
Should this happen as an explicit call from somewhere in the initialization code? It feels a bit weird to have this be a side effect of importing this file

Reply:
Yeah I guess that would be better, I wanted to keep everything centralized but the import case does make that a bit awkward.